src/classes/REL_Relationships_Con_TDTM.cls

```diff
@@ -35,119 +35,221 @@
+    * @param oldList The list of objects to be updated
+    * @return A DmlWrapper
+    ********************************************************************************************************/ 
+    @TestVisible
+    private DmlWrapper handleAfterUpdate(List<SObject> newList, List<SObject> oldList){
+        DmlWrapper dmlWrapper = new DmlWrapper();
+
+        if (this.isUpdateRecursion() == true){
+            return null;
         }
+
+        this.setUpdateRecursion();
+
+        Map<Id, Contact> newContactsById = new Map<Id, Contact>((List<Contact>)newList);
+        Map<Id, Contact> oldContactsById = new Map<Id, Contact>((List<Contact>)oldList);
         
-        if (!changedGender.isEmpty()) {
-            //query for a map of all relationships where where contact or related contact has changed gender
-            map<id,Relationship__c> mapRelationship = new map<Id,Relationship__c>(
-                [SELECT Id, Contact__c, RelatedContact__c, ReciprocalRelationship__c, Type__c 
-                FROM Relationship__c WHERE Contact__c IN :changedGender OR RelatedContact__c IN :changedGender]
-            );
-            //make a list of "Related Contact" lookup relationships in our set, these we want to update
-            list<Relationship__c> listRelsForUpdate = new list<Relationship__c>();
-            //make a list of mirror relationships for our first list, we need these for the addType method
-            list<Relationship__c> listMirrorRels = new list<Relationship__c>();
-
-            //get the relationships that the gender change affects
-            for (Relationship__c rel : mapRelationship.values()) {
-                //add the changed gender's relationship and reciprocal
-                if (changedGender.contains(rel.RelatedContact__c)) {
-                    listRelsForUpdate.add(rel);
-                    listMirrorRels.add(mapRelationship.get(rel.ReciprocalRelationship__c));
-                }
+        List<MODL_ConRelationshipAutoCreateMapping> contactRelAutoCreateModelsList = this.locateRelationshipAutoCreateMapper().getContactMappingModelsFromSettings();
+
+        MODL_ContactsRelationships contactRelationshipsToCreate = new MODL_ContactsRelationships();
+
+        Map<Id, Contact> newContactRelationshipToUpdateByIdMap = new Map<Id,Contact>();
+
+        for (Contact newContact : newContactsById.values()) {
+            Contact oldContact = oldContactsById.get(newContact.Id);
+            MODL_ContactRelationships contactRelationshipsModel = this.locateRelationshipAutoCreateService().getRelationshipsToCreateForUpdatedContact(newContact, oldContact, contactRelAutoCreateModelsList);
+            
+            if (contactRelationshipsModel.isEmpty() == false) {
+                contactRelationshipsToCreate.addContactRelationshipsModel(contactRelationshipsModel);   
+            }
+                
+            if (this.locateReciprocalRelationshipService().shouldUpdateContactRelationships(newContact, oldContact) == true) {
```

Comment:
@MatthewBlanski I think I figured out whats behind the last test failure. It fails because it tries to throw a null pointer exception first and then the error message does not contain the invalid field label. After some quick debugs, it looks like EDA will first run afterinsert for contact, afterupdate for account and then afterupdate for contact again. For the afterupdate here, oldList is null. I feel like an easy fix is to add guard clause for oldList here in handleAfterUpdate. We can do that way ahead of everything.

Reply:
Also a side note, it is nice to have a unit test method to catch a bug but it feels weird it works this way. Maybe we should consider rework the unit tests for filter too.

Reply:
After adding the guard clause, it looks like an exception was thrown: System.DmlException: Update failed. First exception on row 0 with id 0035600000SmiwjAAB; first error: FIELD_CUSTOM_VALIDATION_EXCEPTION, The field entered as the Filter Field isn't valid. Filter Fields can be any standard or custom field on the object this trigger is for, as well as fields on related records. Make sure you use the field's API name, and omit any EDA namespace prefix ("hed__"). REL_Relationships_Con_TDTM: []. And for some reasons this was not caught by catch block.

Reply:
This is far more strange than that. That error message comes from our TDTM filter, which probably shouldn't even be involved in this scenario. Let's dig into that and see what's happening.

Reply:
I think the reason why it is involved is because it is using `REL_Relationships_Con_TDTM` as the test trigger handler.

Reply:
This is a core EDA bug with oldList becoming null with an invalid filter field., causing unexpected effects. Thank you for finding this in our conversation earlier Xinnan.

Since a bad filter field will prevent the transaction entirely, the end result is the same. It does hide other potential errors at the record level, which is undesired but not critical.

For now the guard clause will solve this matter and we can go and fix the other bug later.